bold text

In [ ]:
import pandas as pd
import numpy as np

Import Data Sets Provided

In [ ]:
GeoTag = pd.read_csv('/content/cells_geo.csv', sep=';')
GeoTag.describe()
GeoTag

Unnamed: 0    VILLES      STATUS  ...          AREA     CELL_ID   SITE_CODE
0              0    ADJAME  In Service  ...           NaN  fbb2e75af3  0bb492ef17
1              1    ADJAME  In Service  ...           NaN  1502501d78  0bb492ef17
2              2    ADJAME  In Service  ...           NaN  0c2e182a0c  0bb492ef17
3              3   ASSINIE  In Service  ...           NaN  69bdfc69fd  8a1e6ce31f
4              4   ASSINIE  In Service  ...           NaN  7a1029d2ab  8a1e6ce31f
...          ...       ...         ...  ...           ...         ...         ...
3969        3969  ZUENOULA  In Service  ...  CENTRE NORTH  1cfc2d7ab1  f91ed8d2e9
3970        3970  ZUENOULA  In Service  ...  CENTRE NORTH  3ae1708153  f91ed8d2e9
3971        3971  ZUENOULA  In Service  ...  CENTRE NORTH  cb71f8a29c  61bee9c70c
3972        3972  ZUENOULA  In Service  ...  CENTRE NORTH  d4a6a0b645  61bee9c70c
3973        3973  ZUENOULA  In Service  ...  CENTRE NORTH  f9f3e42589  61bee9c70c

[3974 rows x 12 columns]

In [ ]:
Telcom1= pd.read_csv('/content/Telcom_dataset.csv')
Telcom1.head()

PRODUTC  VALUE               DATETIME  ... COUNTRY_B     CELL_ID     SITE_ID
0   Voice      0  2012-05-06 23:04:37.0  ...     19e2e  /TJNe+Mmtu  /TJNe+Mmtu
1     sms      0  2012-05-06 23:05:25.0  ...     19e2e  /+cKZKIp41  /+cKZKIp41
2     sms     25  2012-05-06 23:05:41.0  ...     19e2e  /u0FSD+ahi  /u0FSD+ahi
3   Voice     56  2012-05-06 23:05:42.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     14  2012-05-06 23:06:29.0  ...     19e2e  051KhYDCpv  051KhYDCpv

[5 rows x 10 columns]

In [ ]:
Telcom1.describe()

VALUE
count  5001.000000
mean     25.869626
std     101.934502
min       0.000000
25%       0.000000
50%       0.000000
75%      25.000000
max    4440.000000

In [ ]:
Telcom2= pd.read_csv('/content/Telcom_dataset2.csv')
Telcom2.head()

PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B     CELL_ID     SITE_ID
0     sms      0  2012-05-07 23:02:06.0  ...     19e2e  +854AcBQT2  +854AcBQT2
1   Voice      0  2012-05-07 23:03:44.0  ...     19e2e  +854AcBQT2  +854AcBQT2
2     sms      0  2012-05-07 23:04:06.0  ...     19e2e  +laSrk7g6q  +laSrk7g6q
3   Voice     37  2012-05-07 23:04:48.0  ...     19e2e  /xvRrCVKoQ  /xvRrCVKoQ
4     sms     25  2012-05-07 23:04:58.0  ...     19e2e  +bO+qkSonO  +bO+qkSonO

[5 rows x 10 columns]

In [ ]:
Telcom2.describe()

VALUE
count  5001.000000
mean     24.334933
std      97.471586
min       0.000000
25%       0.000000
50%       0.000000
75%      25.000000
max    3380.000000

In [ ]:
Telcom3= pd.read_csv('/content/Telcom_dataset3.csv')
Telcom3.head()

PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B      CELLID     SIET_ID
0   Voice     61  2012-05-08 23:01:28.0  ...     19e2e  +38u2u/rfx  +38u2u/rfx
1     sms      0  2012-05-08 23:01:31.0  ...     19e2e  +Fs4mTvdKx  +Fs4mTvdKx
2   Voice     10  2012-05-08 23:02:59.0  ...     19e2e  +ZaaZozy/+  +ZaaZozy/+
3     sms      0  2012-05-08 23:03:04.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     22  2012-05-08 23:03:16.0  ...     19e2e  /eeMklgyJA  /eeMklgyJA

[5 rows x 10 columns]

In [ ]:
Telcom3.describe()

VALUE
count   5001.000000
mean      26.441512
std      224.200217
min        0.000000
25%        0.000000
50%        0.000000
75%       25.000000
max    12900.000000

Merge the three Telcom Datasets


In [ ]:
#Correct the column names before merge
Telcom1.columns = ['PRODUCT','VALUE','DATE-TIME','CELL ON SITE','DW_A_NUMBER_INT','DW_B_NUMBER_INT','COUNTRY_A','COUNTRY_B','CELL_ID','SITE_ID']
Telcom1.head()

PRODUCT  VALUE              DATE-TIME  ... COUNTRY_B     CELL_ID     SITE_ID
0   Voice      0  2012-05-06 23:04:37.0  ...     19e2e  /TJNe+Mmtu  /TJNe+Mmtu
1     sms      0  2012-05-06 23:05:25.0  ...     19e2e  /+cKZKIp41  /+cKZKIp41
2     sms     25  2012-05-06 23:05:41.0  ...     19e2e  /u0FSD+ahi  /u0FSD+ahi
3   Voice     56  2012-05-06 23:05:42.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     14  2012-05-06 23:06:29.0  ...     19e2e  051KhYDCpv  051KhYDCpv

[5 rows x 10 columns]

In [ ]:
Telcom2.columns =['PRODUCT','VALUE','DATE-TIME','CELL ON SITE','DW_A_NUMBER_INT','DW_B_NUMBER_INT','COUNTRY_A','COUNTRY_B','CELL_ID','SITE_ID']
Telcom2.head()

PRODUCT  VALUE              DATE-TIME  ... COUNTRY_B     CELL_ID     SITE_ID
0     sms      0  2012-05-07 23:02:06.0  ...     19e2e  +854AcBQT2  +854AcBQT2
1   Voice      0  2012-05-07 23:03:44.0  ...     19e2e  +854AcBQT2  +854AcBQT2
2     sms      0  2012-05-07 23:04:06.0  ...     19e2e  +laSrk7g6q  +laSrk7g6q
3   Voice     37  2012-05-07 23:04:48.0  ...     19e2e  /xvRrCVKoQ  /xvRrCVKoQ
4     sms     25  2012-05-07 23:04:58.0  ...     19e2e  +bO+qkSonO  +bO+qkSonO

[5 rows x 10 columns]

In [ ]:
Telcom3.columns=['PRODUCT','VALUE','DATE-TIME','CELL ON SITE','DW_A_NUMBER_INT','DW_B_NUMBER_INT','COUNTRY_A','COUNTRY_B','CELL_ID','SITE_ID']
Telcom3.head()

PRODUCT  VALUE              DATE-TIME  ... COUNTRY_B     CELL_ID     SITE_ID
0   Voice     61  2012-05-08 23:01:28.0  ...     19e2e  +38u2u/rfx  +38u2u/rfx
1     sms      0  2012-05-08 23:01:31.0  ...     19e2e  +Fs4mTvdKx  +Fs4mTvdKx
2   Voice     10  2012-05-08 23:02:59.0  ...     19e2e  +ZaaZozy/+  +ZaaZozy/+
3     sms      0  2012-05-08 23:03:04.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     22  2012-05-08 23:03:16.0  ...     19e2e  /eeMklgyJA  /eeMklgyJA

[5 rows x 10 columns]

In [ ]:
#Merge the three datasets 
Telecom = pd.concat([Telcom1,Telcom2,Telcom3],ignore_index=True)
Telecom.head()

PRODUCT  VALUE              DATE-TIME  ... COUNTRY_B     CELL_ID     SITE_ID
0   Voice      0  2012-05-06 23:04:37.0  ...     19e2e  /TJNe+Mmtu  /TJNe+Mmtu
1     sms      0  2012-05-06 23:05:25.0  ...     19e2e  /+cKZKIp41  /+cKZKIp41
2     sms     25  2012-05-06 23:05:41.0  ...     19e2e  /u0FSD+ahi  /u0FSD+ahi
3   Voice     56  2012-05-06 23:05:42.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     14  2012-05-06 23:06:29.0  ...     19e2e  051KhYDCpv  051KhYDCpv

[5 rows x 10 columns]

In [ ]:
Telecom.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15003 entries, 0 to 15002
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   PRODUCT          15003 non-null  object
 1   VALUE            15003 non-null  int64 
 2   DATE-TIME        15003 non-null  object
 3   CELL ON SITE     15003 non-null  object
 4   DW_A_NUMBER_INT  15003 non-null  object
 5   DW_B_NUMBER_INT  15003 non-null  object
 6   COUNTRY_A        15003 non-null  object
 7   COUNTRY_B        15003 non-null  object
 8   CELL_ID          15003 non-null  object
 9   SITE_ID          13006 non-null  object
dtypes: int64(1), object(9)
memory usage: 1.1+ MB


Clean The GeoTag Dataset

In [ ]:
GeoTag.columns

Index(['Unnamed: 0', 'VILLES', 'STATUS', 'LOCALISATION', 'DECOUPZONE',
       'ZONENAME', 'LONGITUDE', 'LATITUDE', 'REGION', 'AREA', 'CELL_ID',
       'SITE_CODE'],
      dtype='object')

In [ ]:
GeoTag.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3974 entries, 0 to 3973
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Unnamed:0     3974 non-null   int64  
 1   VILLES        3974 non-null   object 
 2   STATUS        3907 non-null   object 
 3   LOCALISATION  3974 non-null   object 
 4   DECOUPZONE    3974 non-null   object 
 5   ZONE_NAME     3968 non-null   object 
 6   LONGITUDE     3974 non-null   float64
 7   LATITUDE      3974 non-null   float64
 8   REGION        3974 non-null   object 
 9   AREA          3951 non-null   object 
 10  CELL_ID       3974 non-null   object 
 11  SITE_ID       3974 non-null   object 
dtypes: float64(2), int64(1), object(9)
memory usage: 372.7+ KB


Replace the missing values with hyphens.

In [ ]:
GeoTag= pd.DataFrame(GeoTag)
GeoTag.fillna('-')

Unnamed:0    VILLES      STATUS  ...          AREA     CELL_ID     SITE_ID
0             0    ADJAME  In Service  ...             -  fbb2e75af3  0bb492ef17
1             1    ADJAME  In Service  ...             -  1502501d78  0bb492ef17
2             2    ADJAME  In Service  ...             -  0c2e182a0c  0bb492ef17
3             3   ASSINIE  In Service  ...             -  69bdfc69fd  8a1e6ce31f
4             4   ASSINIE  In Service  ...             -  7a1029d2ab  8a1e6ce31f
...         ...       ...         ...  ...           ...         ...         ...
3969       3969  ZUENOULA  In Service  ...  CENTRE NORTH  1cfc2d7ab1  f91ed8d2e9
3970       3970  ZUENOULA  In Service  ...  CENTRE NORTH  3ae1708153  f91ed8d2e9
3971       3971  ZUENOULA  In Service  ...  CENTRE NORTH  cb71f8a29c  61bee9c70c
3972       3972  ZUENOULA  In Service  ...  CENTRE NORTH  d4a6a0b645  61bee9c70c
3973       3973  ZUENOULA  In Service  ...  CENTRE NORTH  f9f3e42589  61bee9c70c

[3974 rows x 12 columns]

Remove Duplicates  from the Telecom Dataset

In [ ]:
Telecom= Telecom.drop_duplicates()
Telecom.head()

PRODUCT  VALUE              DATE-TIME  ... COUNTRY_B     CELL_ID     SITE_ID
0   Voice      0  2012-05-06 23:04:37.0  ...     19e2e  /TJNe+Mmtu  /TJNe+Mmtu
1     sms      0  2012-05-06 23:05:25.0  ...     19e2e  /+cKZKIp41  /+cKZKIp41
2     sms     25  2012-05-06 23:05:41.0  ...     19e2e  /u0FSD+ahi  /u0FSD+ahi
3   Voice     56  2012-05-06 23:05:42.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4   Voice     14  2012-05-06 23:06:29.0  ...     19e2e  051KhYDCpv  051KhYDCpv

[5 rows x 10 columns]

In [ ]:
Telecom.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 14750 entries, 0 to 15002
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   PRODUCT          14750 non-null  object
 1   VALUE            14750 non-null  int64 
 2   DATE-TIME        14750 non-null  object
 3   CELL ON SITE     14750 non-null  object
 4   DW_A_NUMBER_INT  14750 non-null  object
 5   DW_B_NUMBER_INT  14750 non-null  object
 6   COUNTRY_A        14750 non-null  object
 7   COUNTRY_B        14750 non-null  object
 8   CELL_ID          14750 non-null  object
 9   SITE_ID          12789 non-null  object
dtypes: int64(1), object(9)
memory usage: 1.2+ MB


In [ ]:
GeoTag.columns

Index(['Unnamed:0', 'VILLES', 'STATUS', 'LOCALISATION', 'DECOUPZONE',
       'ZONE_NAME', 'LONGITUDE', 'LATITUDE', 'REGION', 'AREA', 'CELL_ID',
       'SITE_ID'],
      dtype='object')

Merge the Telecom and GeoTag Datasets

In [ ]:
GeoMobile= Telecom.merge(GeoTag,on='SITE_ID')
GeoMobile

PRODUCT  VALUE  ...           AREA   CELL_ID_y
0         sms      0  ...   CENTRE NORTH  24ccdda3d5
1         sms      0  ...   CENTRE NORTH  ebfc58af70
2         sms      0  ...   CENTRE NORTH  ba8ce6e4ef
3         sms      0  ...   CENTRE NORTH  c2545171dc
4         sms      0  ...   CENTRE NORTH  f15bd5483f
...       ...    ...  ...            ...         ...
50313   Voice     17  ...  ABIDJAN NORTH  c179f33d5e
50314   Voice     17  ...  ABIDJAN NORTH  0a3f751619
50315     sms      0  ...   CENTRE NORTH  0427a26d34
50316     sms      0  ...   CENTRE NORTH  7c72293cf2
50317     sms      0  ...   CENTRE NORTH  be18b5b981

[50318 rows x 21 columns]

In [ ]:
GeoMobile.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 50318 entries, 0 to 50317
Data columns (total 21 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   PRODUCT          50318 non-null  object 
 1   VALUE            50318 non-null  int64  
 2   DATE-TIME        50318 non-null  object 
 3   CELL ON SITE     50318 non-null  object 
 4   DW_A_NUMBER_INT  50318 non-null  object 
 5   DW_B_NUMBER_INT  50318 non-null  object 
 6   COUNTRY_A        50318 non-null  object 
 7   COUNTRY_B        50318 non-null  object 
 8   CELL_ID_x        50318 non-null  object 
 9   SITE_ID          50318 non-null  object 
 10  Unnamed:0        50318 non-null  int64  
 11  VILLES           50318 non-null  object 
 12  STATUS           49615 non-null  object 
 13  LOCALISATION     50318 non-null  object 
 14  DECOUPZONE       50318 non-null  object 
 15  ZONE_NAME        50300 non-null  object 
 16  LONGITUDE        50318 non-null  float64
 17  LATITUDE    

Separation of Date and Time Column

In [ ]:
new = GeoMobile["DATE-TIME"].str.split(" ", n = 1, expand = True) 

In [ ]:
GeoMobile['DATE']= new[0]
GeoMobile

PRODUCT  VALUE  ...   CELL_ID_y        DATE
0         sms      0  ...  24ccdda3d5  2012-05-07
1         sms      0  ...  ebfc58af70  2012-05-07
2         sms      0  ...  ba8ce6e4ef  2012-05-07
3         sms      0  ...  c2545171dc  2012-05-07
4         sms      0  ...  f15bd5483f  2012-05-07
...       ...    ...  ...         ...         ...
50313   Voice     17  ...  c179f33d5e  2012-05-09
50314   Voice     17  ...  0a3f751619  2012-05-09
50315     sms      0  ...  0427a26d34  2012-05-09
50316     sms      0  ...  7c72293cf2  2012-05-09
50317     sms      0  ...  be18b5b981  2012-05-09

[50318 rows x 22 columns]

In [ ]:
GeoMobile['TIME']= new[1]

In [ ]:
GeoMobile.drop(columns =["DATE-TIME"], inplace = True) 
GeoMobile

PRODUCT  VALUE CELL ON SITE  ...   CELL_ID_y        DATE        TIME
0         sms      0            1  ...  24ccdda3d5  2012-05-07  00:00:00.0
1         sms      0            1  ...  ebfc58af70  2012-05-07  00:00:00.0
2         sms      0            1  ...  ba8ce6e4ef  2012-05-07  00:00:00.0
3         sms      0            1  ...  c2545171dc  2012-05-07  00:00:00.0
4         sms      0            1  ...  f15bd5483f  2012-05-07  00:00:00.0
...       ...    ...          ...  ...         ...         ...         ...
50313   Voice     17            4  ...  c179f33d5e  2012-05-09  00:01:42.0
50314   Voice     17            4  ...  0a3f751619  2012-05-09  00:01:42.0
50315     sms      0            2  ...  0427a26d34  2012-05-09  00:01:43.0
50316     sms      0            2  ...  7c72293cf2  2012-05-09  00:01:43.0
50317     sms      0            2  ...  be18b5b981  2012-05-09  00:01:43.0

[50318 rows x 22 columns]

#ANALYSIS

Most visited  City

In [ ]:
GeoMobile.VILLES.describe()

count        50318
unique         400
top       YOPOUGON
freq          8549
Name: VILLES, dtype: object

#The most visited City in Ivory Coast is Yopougon with a frequency of  8549.

Which city has a low coverage area

In [ ]:
GeoMobile['Number of Areas']=1
Numbers =GeoMobile .groupby('VILLES').count().sort_values(by='AREA')

In [ ]:
Coverage = pd.DataFrame(Numbers['Number of Areas'].head())
Coverage

Number of Areas
VILLES                      
KOTOULA                    2
DANANON                    2
DJENIDOUFLA                3
DJOUKOUYO                  3
DAKPADOU                   3

#The City of Kotoula is the city that has the lowest coverage area.

Which city has a high coverage area

In [ ]:
GeoMobile['Number of Areas']=1
Numbers =GeoMobile .groupby('VILLES').count().sort_values(by='AREA', ascending=False)

In [ ]:
Coverage = pd.DataFrame(Numbers['Number of Areas'].head())
Coverage

Number of Areas
VILLES                   
YOPOUGON             8549
ABOBO                6159
COCODY               3716
ADJAME               2655
KOUMASSI             2275

#The City  of Yopougon has the highest coverage area.